drivers/include/at.h

```diff
@@ -186,19 +209,49 @@ typedef struct {
 #endif
 } at_dev_t;
 
+/**
+ * @brief AT device initialization parameters
+*/
+typedef struct {
+    uart_t uart;            /**< UART device where the AT device is attached */
+    uint32_t baudrate;      /**< UART device baudrate */
+    char *rx_buf;           /**< UART rx buffer */
+    size_t rx_buf_size;     /**< UART rx buffer size */
+    /** 
+     * Response parsing buffer - used for classifying DCE responses and holding 
+     * detailed error information. 
+     * If you don't care about URCs (MODULE_AT_URC is undefined) this must only 
+     * be large enough to hold responses like `OK`, `ERROR` or `+CME ERROR: 1234`. 
+     * Otherwise adapt it's size to the maximum length of the URCs you are expecting 
+     * and actually care about. */
+    char *rp_buf;
+    size_t rp_buf_size;     /**< response parsing buffer size */
+} at_dev_init_t;
+
+/**
+ * @brief Get extended error information of the last command sent.
+ *
+ * If a previous at_* method returned with -AT_ERR_EXTENDED, you can retrieve
+ * a pointer to the error string with this.
+ *
+ * @param[in] dev device to operate on
+ *
+ * @retval \0-terminated string containing the error value.
+ */
+static inline char const *at_get_err_info(at_dev_t *dev)
+{
+    return dev->rp_buf;
+}
 /**
  * @brief   Initialize AT device struct
  *
  * @param[in]   dev         struct to initialize
- * @param[in]   uart        UART the device is connected to
- * @param[in]   baudrate    baudrate of the device
- * @param[in]   buf         input buffer
- * @param[in]   bufsize     size of @p buf
+ * @param[in]   init        init struct, may be destroyed after return
  *
  * @retval     success code UART_OK on success
  * @retval     error code UART_NODEV or UART_NOBAUD otherwise
  */
-int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize);
+int at_dev_init(at_dev_t *dev, at_dev_init_t const *init);
```

Comment:
```suggestion
int at_dev_init(at_dev_t *dev, const at_dev_init_t *init);
```
feels more idiomatic

Reply:
I don't know about that. I use *east const* for consistency, e.g. 
```c
char const *const c;
```
is more consistent than
```c
const char *const c;
```
in the sense that `const` *applies to whatever is to the left*